webClient/src/app/editor/code-editor/monaco/hiliters/rexx.ts

```diff
@@ -0,0 +1,184 @@
+
+      // strings
+      [/"([^"\\]|\\.)*$/, 'string.invalid'], // non-teminated string
+			[/'([^'\\]|\\.)*$/, 'string.invalid'], // non-teminated string
+			[/"/, 'string', '@doublestring'],
+			[/'/, 'string', '@singlestring'],
+
+      //procedure
+      //entity.name.function.rexx
+      [/\b[A-Za-z@#$!?_][A-Za-z@#$!?_0-9]*:/, {token: 'constructor'}],
+
+      // numbers
+      [/([0-9]+(\.)?[0-9]*(e[-+]?[0-9]+)?|[0-9]*(\.)?[0-9]+)(e[-+]?[0-9]+)?\b/, 'number'],
+
+      //other constants
+//disabled because the coloring was overbearing
+//      [/[0-9\.][A-Za-z0-9@#$¢.!?_]*/, {token: 'constant.other.rexx'}],
+
+      //operators
+//disabled because theoperator section above root is used instead
+//      [/([\+-/*%&amp;|()¬\\=&lt;&gt;])/, {token: 'operators.rexx'}],
+
+      //call function calls
+      [/(call)(\s+[A-Za-z@#$!?_0-9]+)/, ['keyword', { cases: { '@functions': 'builtin-fcall', '@default': 'fcall'}}]],
+
+      //keywords
+      [/\b(do|forever|while|until|to|by|for|end|exit|if|then|else|iterate|leave|nop|return|select|when|otherwise|call(\s+(off|on)\s+(error|failure(\s+name)?|halt))?|signal(\s+(off|on)\s+(error|failure(\s+name)?|halt|novalue|syntax))|address\s+\w+|arg|drop|interpret|numeric\s+(digits|form(\s+(scientific|engineering|value))?|fuzz)|options|parse(\s+upper)?\s+(external|numeric|source|value|var|version)?|with|procedure(\s+expose)?|pull|push|queue|say|trace\s+\w+|upper)\b(?!\.)/, {token: 'keyword'}],
+
+      //non-call function calls
+      [/\b[A-Za-z@#$!?_0-9]+(?=\()/, { cases: { '@functions': 'builtin-fcall', '@default': 'fcall'}}],
+
+      //variables
+//disabled because the coloring was overbearing
+//      [/\b[A-Za-z@#$¢!?_][A-Za-z0-9@#$¢.!?_]*/, {token: 'variable.name'}],
+
+      // delimiters and operators
+      [/[{}()\[\]]/, '@brackets'],
+
+      //math symbols
+      [/@symbols/, { cases: { '@operators': 'operator' } } ],
+
+      // delimiter: after number because of .\d floats
+      [/[;,.]/, 'delimiter'],
+
+
+//------------------------
+//Section below is made redundant by code above inspired by other open source libraries
+//------------------------
+
+//procedure
+//      [/^[ \t]*[A-Za-z0-9]+\:/, {token: 'constructor'}],
+
+//function calls
+//      [/([A-Za-z0-9.]+)(\()/, [{ cases: { '@functions': 'builtin-fcall', '@default': 'fcall'}}, '@brackets']],
+//      [/(call)(\s+[A-Za-z0-9]+)/, ['keyword', { cases: { '@functions': 'builtin-fcall', '@default': 'fcall'}}]],
+      
+// identifiers and keywords
+//      [/[a-z_$][\w$]*/, { cases: { '@keywords': 'keyword' } }],
+
+// whitespace
+[/[ \t\r\n]+/, ''],
+
+//      [/\/\*/, 'comment', '@comment' ],
+      
+// strings
+//      [/"([^"\\]|\\.)*$/, 'string.invalid'], // non-teminated string
+//			[/'([^'\\]|\\.)*$/, 'string.invalid'], // non-teminated string
+//			[/"/, 'string', '@doublestring'],
+//			[/'/, 'string', '@singlestring'],
+//------------------------
+//end of unused section
```

Comment:
If redundant, why are we keeping it? Is it useful?

Reply:
It also works, but works differently, so it may prove to be a solution to problems later on.